src/crawlee/proxy_configuration.py

```diff
@@ -85,7 +85,7 @@ def __init__(
                 the proxy selection mechanism.
         """
         self._next_custom_url_index = 0
-        self._used_proxy_urls = dict[str, URL]()
+        self._used_proxy_urls = dict[str, Union[URL, None]]()
```

Comment:
Mypy was very aggressive when I wanted to use **|**  on this specific line. I got bullied into using Union

Reply:
Yeah, AFAIK in generics the pipe is still not possible